packages/lit-html/src/lib/directives/unsafe-html.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+// import {isPrimitive} from '../lib/parts.js';
+import {
+  directive,
+  NodePart,
+  Part,
+  Directive,
+  nothing,
+  TemplateResult,
+  noChange,
+} from '../lit-html.js';
+
+/**
+ * Renders the result as HTML, rather than text.
+ *
+ * Note, this is unsafe to use with any user-provided input that hasn't been
+ * sanitized or escaped, as it may lead to cross-site-scripting
+ * vulnerabilities.
+ */
+export const unsafeHTML = directive(
+  class UnsafeHTML extends Directive {
+    value: unknown = nothing;
+    templateResult?: TemplateResult;
+
+    constructor(part: Part) {
+      super();
+      // TODO (justinfagnani): We probably don't want to use instanceof
+      if (!(part instanceof NodePart)) {
+        throw new Error('unsafeHTML can only be used in text bindings');
+      }
+    }
+
+    render(value: unknown) {
```

Comment:
What happens if the value is a TemplateResult? Can we add a test for that?

Reply:
Should we throw? Or throw if values are passed?

Reply:
Seems like we should only allow `string`, frankly. I don't see a reason for passing `nothing` or `noChange` (or any other value type, even non-string primitives) as an argument to the directive; the directive turns strings into HTML; if you want something else, you should do that using conditionals on the outside of the directive. Less code and very hard to imagine this would break someone.

Reply:
Updated to throw on non-strings on the theory that this is usually only used for strings already, so it isn't much of a breaking change. We also don't even import this directive internally.